backend/services/implementations/classService.ts

```diff
@@ -72,6 +72,28 @@ class ClassService implements IClassService {
     return (await this.mapClassToClassDTOs([classObj]))[0];
   }
 
+  async getClassByTestSessionId(
+    testSessionId: string,
+  ): Promise<ClassResponseDTO> {
+    let classes: Class[];
+    try {
+      classes = await MgClass.find({ testSessions: { $eq: testSessionId } });
+      if (!classes.length) {
+        throw new Error(
+          `Class with test session id ${testSessionId} not found`,
+        );
+      } else if (classes.length > 1) {
+        throw new Error(
+          `More than one class has the same Test Session of id ${testSessionId}`,
+        );
+      }
+    } catch (error: unknown) {
+      Logger.error(`Failed to get Class. Reason = ${getErrorMessage(error)}`);
+      throw error;
+    }
+    return (await this.mapClassToClassDTOs(classes))[0];
+  }
+
```

Comment:
created a [ticket](https://www.notion.so/uwblueprintexecs/Update-getClassByTestSession-to-support-completed-tests-852efb2999cb4033884362b9993605e4?pvs=4) on the board to update this function with handling to filter out students who have already taken the test. not adding this logic now, since i think it makes sense to do that after we implement test submission logic! let me know if you think otherwise and i can add that